### PR TITLE
Bugfix: Mismatch in value type of light-sensor-type attribute of Illuminance Measurement Cluster.

### LINKED ISSUE #26556: [XML] Update Illuminance Measurement Cluster XML to match updated spec
The data types names has been updated in Illuminance Measurement Cluster and must be updated in the XML as well.

**Added:**
LightSensorTypeEnum

Spec PR: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/6469 (already merged)

### PR DESCRIPTION
**Pull request by riwaghe**
Fixed Illuminance Measurement Cluster data model using proper enum type, now it uses the correctly defined type.

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]
I am fine with the change. But the table in the spec should be updated to indicate this define type for the LightSensorType attribute instead of enum8
[Illuminance Measurement Attributes](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/IlluminanceMeasurement.adoc#5-attributes)

File: src/app/zap-templates/zcl/data-model/chip/illuminance-measurement-cluster.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@ limitations under the License.
       <attribute side="server" code="0x0001" define="ILLUM_MIN_MEASURED_VALUE" type="INT16U" min="0x0001" max="0xFFFD" writable="false"                   isNullable="true"                  optional="false">MinMeasuredValue</attribute>
       <attribute side="server" code="0x0002" define="ILLUM_MAX_MEASURED_VALUE" type="INT16U" min="0x0002" max="0xFFFE" writable="false"                   isNullable="true"                  optional="false">MaxMeasuredValue</attribute>
       <attribute side="server" code="0x0003" define="ILLUM_TOLERANCE"          type="INT16U" min="0x0000" max="0x0800" writable="false"                                                      optional="true" >Tolerance</attribute>
-      <attribute side="server" code="0x0004" define="ILLUM_LIGHT_SENSOR_TYPE"  type="ENUM8"  min="0x00"   max="0xFF"   writable="false"                   isNullable="true" default="0xFF"   optional="true" >LightSensorType</attribute>
+      <attribute side="server" code="0x0004" define="ILLUM_LIGHT_SENSOR_TYPE"  type="LightSensorType"  min="0x00"   max="0xFF"   writable="false"                   isNullable="true" default="0xFF"   optional="true" >LightSensorType</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/illuminance-measurement-cluster.xml:33]: Please do not make changes to XML that diverge it from the spec.  We've had way too much of this, and then the spec gets changed in a _different_ way and the XML has to get changed again, etc, and every change affects public APIs that people are using, which is bad.

In particular, the spec is likely to _not_ call the type of this thing LightSensorType.

The right ordering here is to get the spec to say the right things, then align the XML here to match.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/zcl/data-model/chip/illuminance-measurement-cluster.xml (hunk 1 above)
Comment (andersesbensen) [reply src/app/zap-templates/zcl/data-model/chip/illuminance-measurement-cluster.xml:33]: But do we really want the spec to explicitly define data type names is that not going too far? the fact the data type is called LightSensorType XML is not a spec violation as I see it, this is just a trait of the XML representation of the spec.

I would argue that the spec is more readable as it is, stating that this is an 8-bit enumeration, and this enumeration is defined in its own section in the spec.

But the fact ENUM8 type is translated into a uint8_t when ZAP is reading the XML files is a little dangerous.

**Comment by ReneJosefsen** [comment]
It might be worth coordinating this effort with the in-flight editorial for this cluster https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/6469 so we do not have to change this name twice?

File: src/app/zap-templates/zcl/data-model/chip/illuminance-measurement-cluster.xml (hunk 1 above)
Comment (mamehta) [reply src/app/zap-templates/zcl/data-model/chip/illuminance-measurement-cluster.xml:33]: @bzbarsky-apple The specification changes are merged - https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/6469

Can this PR be approved now?

**Review by bzbarsky-apple — APPROVED** [review]
Thank you!

**Comment by bzbarsky-apple** [comment]
@riwaghe This needs to be rebased/merged to tip for those generated code file conflicts.  Let me know if you want me to do that, please.

**Pull request merged**